src/Centreon/Infrastructure/PlatformTopology/PlatformTopologyRepositoryRDB.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/*
+ * Copyright 2005 - 2020 Centreon (https://www.centreon.com/)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * For more information : contact@centreon.com
+ *
+ */
+declare(strict_types=1);
+
+namespace Centreon\Infrastructure\PlatformTopology;
+
+use Centreon\Domain\Entity\EntityCreator;
+use Centreon\Domain\PlatformTopology\Interfaces\PlatformTopologyRepositoryInterface;
+use Centreon\Domain\PlatformTopology\PlatformTopology;
+use Centreon\Infrastructure\DatabaseConnection;
+use Centreon\Infrastructure\Repository\AbstractRepositoryDRB;
+
+/**
+ * This class is designed to manage the repository of the platform topology requests
+ *
+ * @package Centreon\Infrastructure\PlatformTopology
+ */
+class PlatformTopologyRepositoryRDB extends AbstractRepositoryDRB implements PlatformTopologyRepositoryInterface
+{
+    /**
+     * PlatformTopologyRepositoryRDB constructor.
+     * @param DatabaseConnection $db
+     */
+    public function __construct(DatabaseConnection $db)
+    {
+        $this->db = $db;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function addPlatformToTopology(PlatformTopology $platformTopology): void
+    {
+        $request = $this->translateDbName('
+            INSERT INTO `:db`.platform_topology (`address`, `name`, `type`, `parent_id`)
+            VALUES (:address, :name, :type, :parentId)
+        ');
+
+        $statement = $this->db->prepare($request);
+        $statement->bindValue(':address', $platformTopology->getAddress(), \PDO::PARAM_STR);
+        $statement->bindValue(':name', $platformTopology->getName(), \PDO::PARAM_STR);
+        $statement->bindValue(':type', $platformTopology->getType(), \PDO::PARAM_STR);
+        $statement->bindValue(':parentId', $platformTopology->getParentId(), \PDO::PARAM_INT);
```

Comment:
You won't be able to bind a NULL on parent_id if the parent adress was not sent in the request :
https://github.com/centreon/centreon/pull/8880/files#diff-dc6d2405163adeac8ba8a49fc0a65bbfR74

Reply:
why we can't bind NULL value here ? If we register a central server, this should be null